models/dps/index.go

```diff
@@ -19,49 +19,18 @@ import (
 	"github.com/onflow/flow-go/model/flow"
 )
 
-type State interface {
-	Index() Index
-	Data() Data
-	Last() Last
-	Height() Height
-	Commit() Commit
-	Raw() Raw
-	Ledger() Ledger
+type IndexReader interface {
+	Last() (uint64, error)
+	Header(height uint64) (*flow.Header, error)
+	Commit(height uint64) (flow.StateCommitment, error)
+	Events(height uint64, types ...flow.EventType) ([]flow.Event, error)
+	Registers(height uint64, paths []ledger.Path) ([]ledger.Value, error)
 }
 
-type Index interface {
+type IndexWriter interface {
 	Header(height uint64, header *flow.Header) error
 	Commit(height uint64, commit flow.StateCommitment) error
-	Payloads(height uint64, paths []ledger.Path, payload []*ledger.Payload) error
 	Events(height uint64, events []flow.Event) error
-	Last(commit flow.StateCommitment) error
-}
-
-type Data interface {
-	Header(height uint64) (*flow.Header, error)
-	Events(height uint64, types ...flow.EventType) ([]flow.Event, error)
-}
-
-type Last interface {
-	Height() uint64
-	Commit() flow.StateCommitment
-}
-
-type Height interface {
-	ForBlock(blockID flow.Identifier) (uint64, error)
-	ForCommit(commit flow.StateCommitment) (uint64, error)
-}
-
-type Commit interface {
-	ForHeight(height uint64) (flow.StateCommitment, error)
-}
-
-type Raw interface {
-	WithHeight(height uint64) Raw
-	Get(key []byte) ([]byte, error)
-}
-
-type Ledger interface {
-	WithVersion(version uint8) Ledger
-	Get(*ledger.Query) ([]ledger.Value, error)
+	Payloads(height uint64, paths []ledger.Path, value []*ledger.Payload) error
+	Last(height uint64) error
```

Comment:
When I see this it makes me feel like those should be called `Reader` and `Writer` and be in a neat little `index` package :p Not sure if it makes sense to separate it from the state though, and pretty sure I wouldn't like to have all of the state stuff in `index` either.